networking/src/NTP/Client.hs

```diff
@@ -39,16 +38,19 @@ import           System.Wlog (LoggerName, WithLogger, logDebug, logError, logInf
                               modifyLoggerName)
 
 import           Mockable.Class (Mockable)
-import           Mockable.Concurrent (Async, Concurrently, Delay, concurrently, forConcurrently,
+import           Mockable (Async, Concurrently, CurrentTime, Delay, concurrently, currentTime, forConcurrently,
                                       timeout, withAsync)
 import           NTP.Packet (NtpPacket (..), evalClockOffset, mkCliNtpPacket, ntpPacketSize)
 import           NTP.Util (createAndBindSock, resolveNtpHost, selectIPv4, selectIPv6,
                            udpLocalAddresses, withSocketsDoLifted)
 
-data NtpClientSettings m = NtpClientSettings
+data NtpStatus = NtpSyncOk | NtpDesync Microsecond
+    deriving (Eq, Show)
+
+data NtpClientSettings = NtpClientSettings
     { ntpServers         :: [String]
       -- ^ list of servers addresses
-    , ntpHandler         :: (Microsecond, Microsecond) -> m ()
+    , ntpStatus          :: TVar (Maybe NtpStatus)
```

Comment:
Ah, well I suppose `Nothing` basically means `NtpStatusUnavailable`.

Reply:
Not really, I use `Nothing` to wait until ntp client fills it with a value. If we had `NtpStatusUnavailable` it would be filled with `Just NtpStatusUnavailable` at the start.

Reply:
Ok, so `Nothing` means "pending". Perhaps we want `Maybe (TVar NtpStatus)` instead? Where `NtpStatusPending` is one of the status constructors?

Then we'd define

```Haskell
currentNtpStatus :: Maybe (TVar NtpStatus) -> UINtpStatus
currentNtpStatus Nothing = UINtpUnavailable
currentNtpStatus (Just tvar) = do
    st <- atomically (readTVar tvar)
    case st of
        NtpDrift (updatedAt :: Time) (drift :: Int) -> UINtpDrift drift
        NtpPending -> UINtpPending
```

What do you think?

Reply:
@avieth I broke the branch with a rebase, and you commented before I restored it.  We have `TVar NtpStatus`, but we  adding `NtpPending` to the `NtpStatus` is a good idea anyway.  Then we can use `NtpStatusUnavailable` when no respond arrived from any of the ntp servers (within the timeout which is now set to `30sec` in `lib/configuration.yaml`). Now we use a preconfigured threshold to deduce `NtpSyncOk` or `NtpDesync Milliseconds`, which is set to `5sec`. It just should be `<< slotDuration` - I think it is better to compute the threshold depending on slot duration and then it's better to use `NtpSyncOK` rather than expose the drift in `Milliseconds` (the wallet might not now the current slot duration).